web-ng/src/app/services/project/project.service.ts

```diff
@@ -71,6 +71,10 @@ export class ProjectService {
     return this.activeProject$;
   }
 
+  getAllProjects$(): Observable<Project[]> {
```

Comment:
We should only fetch projects accessible to the user (`getAccessibleProjects()`)

Reply:
accessible would be projects that the user is owner and contributor of?

Reply:
Accessible in this content would be any project a user can open. That means they have _any_ roles in the project (owners, manager, contributor, or viewer).

Reply:
so that means all the projects would be accessible. 
https://github.com/google/ground-platform/blob/master/web-ng/src/app/shared/converters/firebase-data-converter.ts#L91

Reply:
The code you're pointing to is how the remote deserialization layer handles an unrecognized role string.

The function we're discussing here returns a list of projects that are accessible to the user. Users clearly should not and do not access _all_ projects in the database, only the ones they've been granted access to. For them to be granted access to a project, their email address must appear in the project's ACL.

Reply:
got it so i will check the user email address with the project acl email addresses.

Reply:
updated

Reply:
so this means get data using query instead of filtering the data after fetching?